File: dotnet/console/Console/Scenarios/CancelDocumentScenario.cs
using Lacuna.Signer.Api.Documents;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Console.Scenarios
{
    class CancelDocumentScenario : Scenario
    {
        /**
         * This scenario demonstrates how to cancel a document as an organization application.
         */
        public override async Task RunAsync()
        {
            // 1. You need a document id
            var result = await CreateDocumentAsync();
            var docId = result.DocumentId;

            // 2. Create a cancellation request and give it a reason. Then send the cancellation request
            await SignerClient.CancelDocumentAsync(docId, new CancelDocumentRequest { Reason = "This is a document cancellation" });

        }
    }
}

File: nodejs/scenarios/cancelDocumentScenario.ts
import {
  DocumentsApi,
  DocumentsCancelDocumentRequest,
} from "signer-node-client";
import { config, CreateDocument } from "./scenario";

const documentsApi = new DocumentsApi(config);

// 1 - You need a document id
CreateDocument().then((res) => {
  const documentId = res.documentId;

  //2 - Create a cancellation request and give it a reason
  const cancelDocument: DocumentsCancelDocumentRequest = {
    reason: "This is a document cancellation",
  };

  //3 - Send the cancellation request

  documentsApi
    .apiDocumentsIdCancellationPost(documentId, cancelDocument)
    .then((response) => {
      console.log(response.status);
    });
});
